src/gui/input/HotkeySettings.cpp

```diff
@@ -0,0 +1,363 @@
+#include "gui/input/HotkeySettings.h"
+#include <config/ActiveSettings.h>
+#include <gui/guiWrapper.h>
+#include "input/InputManager.h"
+#include "HotkeySettings.h"
+
+#if BOOST_OS_LINUX || BOOST_OS_MACOS
+#include "resource/embedded/resources.h"
+#endif
+
+extern WindowInfo g_window_info;
+const std::unordered_map<sHotkeyCfg*, std::function<void(void)>> HotkeySettings::s_cfgHotkeyToFuncMap{
+	{&s_cfgHotkeys.toggleFullscreen, [](void) { s_mainWindow->ShowFullScreen(!s_mainWindow->IsFullScreen()); }},
+	{&s_cfgHotkeys.toggleFullscreenAlt, [](void) { s_mainWindow->ShowFullScreen(!s_mainWindow->IsFullScreen()); }},
+	{&s_cfgHotkeys.exitFullscreen, [](void) { s_mainWindow->ShowFullScreen(false); }},
+	{&s_cfgHotkeys.takeScreenshot, [](void) { g_window_info.has_screenshot_request = true; }},
+	{&s_cfgHotkeys.toggleFastForward, [](void) { ActiveSettings::SetTimerShiftFactor((ActiveSettings::GetTimerShiftFactor() < 3) ? 3 : 1); }},
+};
+
+struct HotkeyEntry
+{
+	enum class InputButtonType : wxWindowID {
+		Keyboard,
+		Controller,
+	};
+
+	std::unique_ptr<wxStaticText> name;
+	std::unique_ptr<wxButton> keyInput;
+	std::unique_ptr<wxButton> controllerInput;
+	sHotkeyCfg& hotkey;
+
+	HotkeyEntry(wxStaticText* name, wxButton* keyInput, wxButton* controllerInput, sHotkeyCfg& hotkey)
+		: name(name), keyInput(keyInput), controllerInput(controllerInput), hotkey(hotkey)
+	{
+		keyInput->SetClientData(&hotkey);
+		keyInput->SetId(static_cast<wxWindowID>(InputButtonType::Keyboard));
+		controllerInput->SetClientData(&hotkey);
+		controllerInput->SetId(static_cast<wxWindowID>(InputButtonType::Controller));
+	}
+};
+
+HotkeySettings::HotkeySettings(wxWindow* parent)
+	: wxFrame(parent, wxID_ANY, "Hotkey Settings")
```

Comment:
Make the strings in this function localizable. E.g. `_("Hotkey Settings")`, same for the hotkey row labels